gemfirexd/client/src/main/java/com/pivotal/gemfirexd/internal/client/am/Statement.java

```diff
@@ -2715,6 +2715,11 @@ void parseSqlAndSetSqlModes(String sql) throws SqlException {
             firstToken = getStatementToken(sql, idx + 1);
           }
         }
+        if (firstToken.equalsIgnoreCase("set")) {
+          if (sql.toLowerCase().contains(" schema ")) {
+              this.connection_.setSchemaSql_ = sql;
+          }
+        }
 // GemStone changes END
```

Comment:
As per docs https://db.apache.org/derby/docs/10.4/ref/rrefsqlj32268.html, this can also be "SET [CURRENT] USER" or "SET CURRENT SQLID" -- note that CURRENT is required for SQLID but optional for others.